notebooks/00_quick_start/rbm_movielens.ipynb

```diff
@@ -0,0 +1,885 @@
+    "\n",
+    "- Model initialization: This is where we tell TF how to build the computational graph. The main parameters to specify are the number of hidden units, the number of training epochs and the minibatch size. Other parameters can be optionally tweaked for experimentation, as explained in ... \n",
+    "\n",
+    "- Model fit: This is where we train the model on the data. The method takes two arguments: the training and test set matrices. Note that the model is trained **only** on the training set, the set is used to display the generalization accuracy of the trained model, useful to have a first idea on the quality of training. \n",
+    "\n",
+    "- Model prediction: This is where we generate ratings for the unseen items. Once the model has been trained and we are satisfied with its overall accuracy, we sample new ratings from the learned distribution. In particular, we extract the top_k (e.g. 10) most relevant recommendation according to some predefined score. The prediction is then returned in dataframe format ready to be analysed and deployed.  \n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "inputHidden": false,
+    "outputHidden": false
+   },
+   "outputs": [],
+   "source": [
+    "#First we initialize the model class\n",
+    "model = RBM(hidden_units= 500, training_epoch = 10, minibatch_size= 60, keep_prob=.9,with_metrics =True ,**header)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that the first time the fit method is called it may take longer to return the result. This is due to the fact that TF needs to initialized the GPU session. You will notice that this is not the case when training the algorithm the second or more times.   "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {
+    "inputHidden": false,
+    "outputHidden": false
+   },
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Creating the computational graph\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "training epoch 0 rmse Train 0.898648\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "done training, Training time 1.7138962\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "training epoch 10 rmse Train 0.702141\n",
+      "precision on the train set 0.40427575\n",
```

Comment:
What is `precision` here? In the text it is called "generalization accuracy". And there is also RMSE. It is a little bit confusing....

Reply:
the test set precision is complemetary to the generalization error. To be more precise, the generazation error is the limit of the test set error for a test size going to inifinity. This is used to understand the generalization capababilities of a model, this is why it is being comprared to the train set error. It is simply given by: 1/m Sum_{mu=1}^{m} Sum{i=1}^Nv 1/s(i) I(v-vp = 0)_{mu,i}, where  m = Nusers, Nv = number of items = number of visible units and s(i) is the number of non-zero elements per row. Also, I() is an indicator function, v the input value of the visible unit (either form train or test set) and vp is the one predicted by the algo. The explanation is in the Docstring of `Mrbm_tensorflow.py`, but I will also add it to the deep dive notebook for clarity.